### ISSUE TITLE
bug: stencil hydrate not calling globalScript in version 4.17.0 onwards

### ISSUE THREAD

**Issue by felipeyukio182** [ionic-team/stencil#6002]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

4.17.0

### Current Behavior

I have an angular SSR application and use the stencil hydrate. Until version 4.16.0, I had no problems hydrating my server-side components, but from 4.17.0 onwards the globalScript initialization code stopped working, it seemed that they were not called.

I run the following initialization script from global-script-start.ts:
```
export default function() {
  const customConfig = window?.CustomConfig;
  window.CustomAPI = {
    color: customConfig?.color || "#003DA5",
  }
  const api = window.CustomAPI;
  window.document.documentElement.style.setProperty("--stencil-custom-color", api.color);
  if (customConfig && typeof customConfig.onReady === 'function') customConfig.onReady(api);
}
```
In stencil.config:
`globalScript: 'src/global-script-start.ts',`

When I run my SSR application, I call the "renderToString" function of the stencil hydrate and send the result to the client. But even though the html is hydrated, this initialization script appears not to have been called, and the css variable "--stencil-custom-color" is not set.

Before version 4.17.0, this script ran without problems and the CSS variable "--stencil-custom-color" was included with the `<html>` tag.

I checked the changelog from 4.16.0 to 4.17.0, and noticed that bundler was migrated from rollup to esbuild.
Could there be something related to this? Or am I missing something?

### Expected Behavior

When I call the renderToString function, the script inside `global-script-start.ts` should be executed, and the CSS variable "--stencil-custom-color" should be set inside the `<html>` tag of the renderToString response.

### System Info

```shell
System: node 20.13.1
    Platform: windows (10.0.19045)
   CPU Model: Intel(R) Core(TM) i7-10510U CPU @ 1.80GHz (8 cpus)
    Compiler: C:\Users\felipe.fupunaga\Documents\Projetos\outros\stencil-hydration-global-script-test\node_modules\@stencil\core\compiler\stencil.js
       Build: 1724698030
     Stencil: 4.21.0
  TypeScript: 5.5.3
      Rollup: 2.56.3
      Parse5: 7.1.2
      jQuery: 4.0.0-pre
      Terser: 5.31.1
```

### Steps to Reproduce

Run the following:
```
git clone https://github.com/felipeyukio182/stencil-hydration-global-script-test.git
cd stencil-hydration-global-script-test
npm i
npm run build
cd packages/angular-ssr-project
npm run serve:ssr
```

Then access http://localhost:4000/ with the network tab open and compare the html preview coming from the server with the html rendered on the page.
The client-side html will have the CSS variable "--stencil-custom-color" (text with green color) while the server-side html will not have any css variable in the `<html>` tag (text with red color). You can also see the blinking effect from red to green.

For comparison, I created the stencil-v4.16.0 branch, where it is possible to run the same test. In this branch, both the client-side and server-side responses contain the css variable "--stencil-custom-color" and both responses contain green text.

### Code Reproduction URL

https://github.com/felipeyukio182/stencil-hydration-global-script-test

### Additional Information

When i run the **main branch**, this is the response:
![image](https://github.com/user-attachments/assets/c09015c1-34d9-442a-808f-82d4ae9bd4d8)

When i run the **stencil-v4.16.0 branch**, the html from server-side and client-side are the same:
![image](https://github.com/user-attachments/assets/91536116-8153-42b5-9f99-021f8609f94c)

And the html from server have the "--stencil-custom-color" CSS variable:
![image](https://github.com/user-attachments/assets/551673ed-cbdd-4f60-865e-4147481cc94b)

**Comment by christian-bromann** [comment]
@felipeyukio182 thanks for raising an issue and providing a reproducible example. Our team currently has a lot of competing priorities and we can't say when we are able to take a look at this. We would appreciate any support in form of investigation on the root cause or actual fixes. We are happy to answer any technical questions as they arise.

**Comment by christian-bromann** [comment]
A fix for this has been released in Stencil `v4.22.0` 🎉
